### PR TITLE
[MPS] Add native implementation for shift ops

### PR DESCRIPTION
**Pull request by malfet**
Similar to how AND/OR/XOR ops are implemented

TODO: Consider using MPS method calls rather than metal kernels

### CODE REVIEW

File: aten/src/ATen/mps/MPSFallback.mm [hunk 1]
```diff
@@ -85,8 +85,6 @@ static Tensor slow_conv2d_forward_mps(const Tensor& self,
   // These ops are not supported via MPS backend currently, and we fallback to run on CPU.
   // For the rest of unsupported ops the user needs to pass 'PYTORCH_ENABLE_MPS_FALLBACK=1'
   // to fallback on CPU, otherwise we will error out.
-  m.impl("bitwise_left_shift.Tensor_out", torch::CppFunction::makeFromBoxedFunction<&mps_fallback>());
-  m.impl("bitwise_right_shift.Tensor_out", torch::CppFunction::makeFromBoxedFunction<&mps_fallback>());
```
Comment (manuelcandales) [inline aten/src/ATen/mps/MPSFallback.mm:89]: If there was a fallback in place, why were things not working before?
Granted that the error was with `__left__.Scalar`. Not sure how that magic works

File: aten/src/ATen/native/native_functions.yaml [hunk 2]
```diff
@@ -8457,29 +8457,29 @@
   device_check: NoCheck   # TensorIterator
   variants: method, function
   dispatch:
-    CPU, CUDA: __lshift__
+    CPU, CUDA, MPS: __lshift__
```
Comment (manuelcandales) [inline aten/src/ATen/native/native_functions.yaml:8460]: Just curious, where is the magic that ties `__lshift__` to `bitwise_left_shift_out`

**Review by manuelcandales — APPROVED** [review]

File: aten/src/ATen/mps/MPSFallback.mm (hunk 1 above)
Comment (manuelcandales) [reply aten/src/ATen/mps/MPSFallback.mm:89]: Oh, the fallback was for the Tensor variant, not for the Scalar ones

Comment (malfet) [reply aten/src/ATen/mps/MPSFallback.mm:89]: Yeah, this only works if one calls `torch.bitwise_left_shift`, not sure what's the story with all those

File: aten/src/ATen/native/native_functions.yaml (hunk 2 above)
Comment (malfet) [reply aten/src/ATen/native/native_functions.yaml:8460]: Here:
https://github.com/pytorch/pytorch/blob/3d7c424a750a1115bfbe56baf0f03b8a05e6ffe9/aten/src/ATen/native/BinaryOps.cpp#L1292

**Pull request closed without merging**